dio/lib/src/failed_request_interceptor.dart

```diff
@@ -0,0 +1,21 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+class FailedRequestInterceptor extends Interceptor {
+  FailedRequestInterceptor({Hub? hub}) : _hub = hub ?? HubAdapter();
+
+  final Hub _hub;
+
+  @override
+  void onError(
+    DioError err,
+    ErrorInterceptorHandler handler,
+  ) {
+    _hub.captureException(err);
```

Comment:
Should we rather convert the `DioError` into SentryEvent with its meta data? e.g. `SentryRequest`, `Mechanism`, etc, similar to `FailedRequestClientAdapter`, I believe it still makes sense to offer the `maxRequestBodySize`, `failedRequestStatusCodes`, and `sendDefaultPii`

Reply:
`failedRequestStatusCodes` is done by `ValidateStatus` from dio. If the status code is bad, Dio throws an exception which the interceptor reports.